keepassxc-browser/content/keepassxc-browser.js

```diff
@@ -843,6 +805,61 @@ kpxc.init = function() {
     initcb();
 };
 
+// Detects DOM changes in the document
+kpxc.initObserver = function() {
+    const observer = new MutationObserver(function(mutations, obs) {
+        if (document.visibilityState === 'hidden' || kpxcUI.mouseDown) {
+            return;
+        }
+
+        // Limit the mutation handling
+        if (mutations.length > _maximumMutations) {
+            return mutations.slice(0, _maximumMutations);
```

Comment:
Do you want to return here? I would think you would want to do `mutations = mutations.slice(0, max)` or just keep a count in your for loop below and break after max mutations.

Reply:
Oops. No return here. My bad. I'd rather just slice the array because it's rare that the maximum amount here is even met. So this is much faster than always calculate the loop count.